### PR TITLE
Fail eagerly to wait for the idle status if we have reached a "final" status

### PR DESCRIPTION
**Pull request by msftristew**
* Rename `wait_for_status()` to `wait_for_idle()` and adjust parameter list appropriately.  If we ever need to wait for a status besides idle (seems unlikely?), then this can be refactored back out again.

* Throw a `LivyUnexpectedError` if the Livy session goes into error or dead.

### CODE REVIEW

**Comment by msftristew** [comment]
1. Output will look similar to when a `LivyClientTimeoutError` is thrown.

2. I am unsure a) if we can get useful error info from Livy when we get this kind of error and b) how we might capture that useful error info and display it to the user.  This merits further discussion.  However, the current error message is "good enough" IMO.

File: remotespark/livyclientlib/livyunexpectederror.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class LivyUnexpectedError(Exception):
+    """An exception that will be shown if some unexpected error happens on the Livy side."""
```
Comment (aggFTW) [inline remotespark/livyclientlib/livyunexpectederror.py]: Can we rename to LivyUnexpectedFinalStatusErrors? This seems too broad.

**Comment by aggFTW** [comment]
If exception is in error, session will have an extra field in the json payload that comes back. Please include that for errored sessions.

File: remotespark/livyclientlib/livyunexpectederror.py (hunk 1 above)
Comment (msftristew) [inline remotespark/livyclientlib/livyunexpectederror.py]: My idea was that this exception could be used for other uses besides just an unexpected final status but this is OK too.

**Comment by aggFTW** [comment]
Please include Closes #66 when addressing a bug. That will close it automatically on merge.

**Comment by msftristew** [comment]
What is the name of the extra field?

**Comment by aggFTW** [comment]
Lgtm! Merging

**Pull request merged**